torch/_tensor.py

```diff
@@ -209,8 +209,18 @@ def __deepcopy__(self, memo):
             return new_tensor
 
     def __reduce_ex__(self, proto):
+        skip_data = getattr(torch.serialization._serialization_tls, "skip_data", False)
+        materialize_fake_tensors = getattr(
+            torch.serialization._serialization_tls, "materialize_fake_tensors", False
+        )
         state = torch._utils._get_obj_state(self)
-        if type(self) is Tensor and not state:
+        # Ignore all state when using FakeTensor with skip_data(materialize_fake_tensors) because FakeTensor has
+        # some state that cannot be pickled
+        if (
+            type(self) is torch._subclasses.fake_tensor.FakeTensor
+            and skip_data
+            and materialize_fake_tensors
```

Comment:
Don't you have materialize_fake_tensors => skip_data? I think you could simplify a lot of this code with this assumption.

Reply:
Done